pkg/controller/node/ipam/sync/sync.go

```diff
@@ -0,0 +1,359 @@
+			"Cannot sync from cloud in mode %q", sync.mode)
+		return fmt.Errorf("cannot sync from cloud in mode %q", sync.mode)
+	}
+
+	glog.V(2).Infof("Updating node spec with alias range, node.PodCIDR = %v", aliasRange)
+
+	if err := sync.set.Occupy(aliasRange); err != nil {
+		glog.Errorf("Error occupying range %v for node %v", aliasRange, sync.nodeName)
+		return err
+	}
+
+	if err := sync.kubeAPI.UpdateNodePodCIDR(ctx, node, aliasRange); err != nil {
+		glog.Errorf("Could not update node %q PodCIDR to %v: %v", node.Name, aliasRange, err)
+		return err
+	}
+
+	glog.V(2).Infof("Node %q PodCIDR set to %v", node.Name, aliasRange)
+
+	if err := sync.kubeAPI.UpdateNodeNetworkUnavailable(node.Name, false); err != nil {
+		glog.Errorf("Error setting route status for node %q: %v", node.Name, err)
+		return err
+	}
+
+	if err := sync.kubeAPI.UpdateNodeNetworkUnavailable(node.Name, false); err != nil {
+		glog.Errorf("Could not update node NetworkUnavailable status to false: %v", err)
+		return err
+	}
+
+	glog.V(2).Infof("Updated node %q PodCIDR from cloud alias %v", node.Name, aliasRange)
+
+	return nil
+}
+
+// updateAliasFromNode updates the cloud alias given the node allocation.
+func (op *updateOp) updateAliasFromNode(ctx context.Context, sync *NodeSync, node *v1.Node) error {
+	if sync.mode != SyncFromCluster {
+		sync.kubeAPI.EmitNodeWarningEvent(
+			node.Name, InvalidModeEvent, "Cannot sync to cloud in mode %q", sync.mode)
+		return fmt.Errorf("cannot sync to cloud in mode %q", sync.mode)
+	}
+
+	_, aliasRange, err := net.ParseCIDR(node.Spec.PodCIDR)
+	if err != nil {
+		glog.Errorf("Could not parse PodCIDR (%q) for node %q: %v",
+			node.Spec.PodCIDR, node.Name, err)
+		return err
+	}
+
+	if err := sync.set.Occupy(aliasRange); err != nil {
+		glog.Errorf("Error occupying range %v for node %v", aliasRange, sync.nodeName)
+		return err
+	}
+
+	if err := sync.cloudAlias.AddAlias(ctx, node.Name, aliasRange); err != nil {
+		glog.Errorf("Could not add alias %v for node %q: %v", aliasRange, node.Name, err)
+		return err
+	}
+
+	if err := sync.kubeAPI.UpdateNodeNetworkUnavailable(node.Name, false); err != nil {
+		glog.Errorf("Could not update node NetworkUnavailable status to false: %v", err)
+		return err
+	}
+
+	glog.V(2).Infof("Updated node %q cloud alias with node spec, node.PodCIDR = %v",
+		node.Name, node.Spec.PodCIDR)
+
+	return nil
+}
+
+// allocateRange allocates a new range and updates both the cloud
+// platform and the node allocation.
+func (op *updateOp) allocateRange(ctx context.Context, sync *NodeSync, node *v1.Node) error {
+	if sync.mode != SyncFromCluster {
+		sync.kubeAPI.EmitNodeWarningEvent(node.Name, InvalidModeEvent,
+			"Cannot allocate CIDRs in mode %q", sync.mode)
+		return fmt.Errorf("controller cannot allocate CIDRS in mode %q", sync.mode)
+	}
+
+	cidrRange, err := sync.set.AllocateNext()
+	if err != nil {
+		return err
+	}
+	// If addAlias returns an hard error, cidrRange will be is leaked as there
```

Comment:
nit: " a hard error"
nit: will be leaked (remove is)

Reply:
done